src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentTrainEnumerator.cs

```diff
@@ -41,56 +61,98 @@ public ReadOnlySpan<byte> AnalyzeSingleTerm(int fieldId, ReadOnlySpan<byte> valu
 
         public void WriteNull(int fieldId, string path)
         {
-            _allocator.From(global::Corax.Constants.NullValueSlice.AsSpan(), out var b);
-            _terms.Add((fieldId, path, b));
+            // nothing to do here
         }
 
         public void Write(int fieldId, ReadOnlySpan<byte> value)
         {
             if (value.Length == 0)
                 return;
-            _allocator.From(value, out var b);
-            _terms.Add((fieldId, null, b));
+
+            if (_mapping.TryGetByFieldId(fieldId, out var field) == false)
+                return;
+
+            // Ensure that we will have enough space to write the content plus an space. 
+            int startLocation = 0;
+            int maxSize = Math.Min(128, value.Length);
+
+            if (value.Length > maxSize)
+            {
+                // The value is too big, therefore we will just select 128 bytes out from a random place.
+                startLocation = Random.Shared.Next(value.Length - maxSize);
+            }
+
+            ProcessSelectedStream(field.Analyzer ?? _lowercaseAnalyzer, value.Slice(startLocation, maxSize));
         }
 
         public void Write(int fieldId, string path, ReadOnlySpan<byte> value)
         {
-            if (value.Length == 0)
-                return;
-            _allocator.From(value, out var b);
-            _terms.Add((fieldId, path, b));
+           Write(fieldId, value);
         }
 
         public void Write(int fieldId, string path, string value)
         {
             if (value.Length == 0)
                 return;
-            _allocator.From(value, out var b);
-            _terms.Add((fieldId, path, b));
+
+            if (_mapping.TryGetByFieldId(fieldId, out var field) == false)
+                return;
+
+            // Ensure that we will have enough space to write the content plus an space. 
+            int startLocation = 0;
+            int maxSize = Math.Min(128, value.Length);
+
+            if (value.Length > maxSize)
+            {
+                // The value is too big, therefore we will just select 128 bytes out from a random place.
+                startLocation = Random.Shared.Next(value.Length - maxSize);
+            }
+
+            // We select only the part we are gonna use, we transcode into UTF8 and then process the stream.
+            Utf8.FromUtf16(value.AsSpan(startLocation, maxSize), _words, out _, out var bytesWritten);
```

Comment:
This is wrong.
Consider what happens when we index 128 emojis, each one is 4 bytes, so you'll need 512 bytes, but `_words` is just 256.

Reply:
Doesn't matter, that's why `bytesWritten` is there. It will never fail and we will still be somewhat on target.